src/HDFFile.cxx

```diff
@@ -141,141 +279,21 @@ int HDFFile::init(std::string filename,
-          auto &a = mem->value;
-          if (a.IsObject()) {
-            for (auto &at : a.GetObject()) {
-              if (at.value.IsString()) {
-                write_attribute_str(se.nxv, at.name.GetString(),
-                                    at.value.GetString());
-              }
-              if (at.value.IsInt64()) {
-                write_attribute(se.nxv, at.name.GetString(),
-                                at.value.GetInt64());
-              }
-              if (at.value.IsDouble()) {
-                write_attribute(se.nxv, at.name.GetString(),
-                                at.value.GetDouble());
-              }
-            }
-          }
-        }
-
-        // TODO
-        // Check if there are NX_attributes and set them
-
-        se.basics = true;
-      }
-
-      bool have_child = false;
-      if (se.jsv->IsObject()) {
-        while (!have_child && se.itr != se.jsv->MemberEnd()) {
-          if (strncmp("NX_", se.itr->name.GetString(), 3) != 0) {
-            if (se.itr->value.IsObject()) {
-              stack.emplace_back(se.itr->name.GetString(), &se.itr->value,
-                                 se.nxv);
-              have_child = true;
-            } else if (se.itr->value.IsString()) {
-              // Treated as a basic scalar string "nexus field" aka "hdf
-              // dataset"
-              string s1 = se.itr->value.GetString();
-              auto dsp = H5Screate(H5S_SCALAR);
-              H5Tset_size(strfix, s1.size());
-              auto ds = H5Dcreate2(se.nxv, se.itr->name.GetString(), strfix,
-                                   dsp, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
-              H5Dwrite(ds, strfix, H5S_ALL, H5S_ALL, H5P_DEFAULT, s1.data());
-              H5Dclose(ds);
-              H5Sclose(dsp);
-            } else if (se.itr->value.IsInt64()) {
-              int64_t val = se.itr->value.GetInt64();
-              auto dsp = H5Screate(H5S_SCALAR);
-              auto ds =
-                  H5Dcreate2(se.nxv, se.itr->name.GetString(), H5T_NATIVE_INT64,
-                             dsp, H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
-              H5Dwrite(ds, H5T_NATIVE_INT64, H5S_ALL, H5S_ALL, H5P_DEFAULT,
-                       &val);
-              H5Dclose(ds);
-              H5Sclose(dsp);
-            } else if (se.itr->value.IsDouble()) {
-              double val = se.itr->value.GetDouble();
-              auto dsp = H5Screate(H5S_SCALAR);
-              auto ds = H5Dcreate2(se.nxv, se.itr->name.GetString(),
-                                   H5T_NATIVE_DOUBLE, dsp, H5P_DEFAULT,
-                                   H5P_DEFAULT, H5P_DEFAULT);
-              H5Dwrite(ds, H5T_NATIVE_DOUBLE, H5S_ALL, H5S_ALL, H5P_DEFAULT,
-                       &val);
-              H5Dclose(ds);
-              H5Sclose(dsp);
-            }
-          }
-          ++se.itr;
-        }
-      }
+  }
 
-      if (!have_child) {
-        stack.pop_back();
-      }
+  while (stack.size() > 0) {
```

Comment:
Use the `empty` method to test emptiness, rather than comparing size with 0: `while (!stack.empty()) {`